llama-index-core/llama_index/core/utils.py

```diff
@@ -57,12 +57,12 @@ def __init__(self) -> None:
 
         # ensure access to data is there
         try:
-            nltk.data.find("corpora/stopwords", paths=[self._nltk_data_dir])
```

Comment:
llama-index-core is bundled with nltk and tiktoken files, so that you don't have to download anything at runtime 

By removing paths here, we are guaranteeing that download is called no? That seems like a waste?

Reply:
@logan-markewich : Thanks a lot for your effort in reviewing this PR! Much appreciated.

By removing the paths here we are actually taking into consideration the whole chain included in `nltk.data.path`. It will not impose the download. On the contrary. It gives us quite a few more chances that the download will not be called at all as you perform the lookup on all the defaults locations + the one that comes with  llama-index-core bundle. Therefore it is not a waste at all. It just removes the assumptions.

What you've added in the following block will actually more sense after my fix:
```
if self._nltk_data_dir not in nltk.data.path:
            nltk.data.path.append(self._nltk_data_dir)
```

The lookup(`nltk.data.find`) assumes that if you don't pass the `paths`, it relies on what is stored in `nltk.data.path`. The moment you pass a different value it changes this logic.

If you don't believe me it's easy to test it :) 

Regular nltk - no llama_index:
```
>>> import nltk
>>> print(nltk.data.path)
['/home/wherka/nltk_data', '/home/wherka/workspace/local_llms/.venv/nltk_data', '/home/wherka/workspace/local_llms/.venv/share/nltk_data', '/home/wherka/workspace/local_llms/.venv/lib/nltk_data', '/usr/share/nltk_data', '/usr/local/share/nltk_data', '/usr/lib/nltk_data', '/usr/local/lib/nltk_data']

```

With llama_index imported and my fix applied:
```
>>> import nltk
>>> from llama_index.core import utils
>>> print(nltk.data.path)
['/home/wherka/nltk_data', '/home/wherka/workspace/local_llms/.venv/nltk_data', '/home/wherka/workspace/local_llms/.venv/share/nltk_data', '/home/wherka/workspace/local_llms/.venv/lib/nltk_data', '/usr/share/nltk_data', '/usr/local/share/nltk_data', '/usr/lib/nltk_data', '/usr/local/lib/nltk_data', '/home/wherka/workspace/local_llms/.venv/lib/python3.10/site-packages/llama_index/core/_static/nltk_cache']
>>> nltk.data.find("corpora/stopwords")
FileSystemPathPointer('/home/wherka/workspace/local_llms/.venv/lib/python3.10/site-packages/llama_index/core/_static/nltk_cache/corpora/stopwords')
>>> nltk.data.find("tokenizers/punkt")
FileSystemPathPointer('/home/wherka/workspace/local_llms/.venv/lib/python3.10/site-packages/llama_index/core/_static/nltk_cache/tokenizers/punkt/PY3')
>>> 
```

I can assure you it does not break anything. It gives us more flexibility.